bindings/wasm/src/did/wasm_core_document.rs

```diff
@@ -40,23 +45,40 @@ use proc_typescript::typescript;
 use wasm_bindgen::prelude::*;
 use wasm_bindgen::JsCast;
 
+pub(crate) struct CoreDocumentLock(tokio::sync::RwLock<CoreDocument>);
+impl CoreDocumentLock {
+  pub(crate) fn new(input: CoreDocument) -> Self {
+    Self(tokio::sync::RwLock::new(input))
+  }
+
+  pub(crate) fn blocking_read(&self) -> impl Deref<Target = CoreDocument> + '_ + Into<ImportedDocumentReadGuard<'_>> {
+    self.0.blocking_read()
+  }
+
+  pub(crate) fn blocking_write(&self) -> impl DerefMut<Target = CoreDocument> + '_ {
+    self.0.blocking_write()
+  }
```

Comment:
So that it will be (relatively) easier to swap out the lock later if we want to. I have tried to hide the lock implementation as much as possible (also internally).

Reply:
Using the concrete type makes reading the other lock stuff easier so I would prefer that. I don't think it'd be very hard to swap the lock.